src/pf-dropdown/pf-dropdown.spec.js

```diff
@@ -0,0 +1,72 @@
+describe("Patternfly Dropdown Component Test", function () {
+  var customElement;
+
+  beforeEach(function (done) {
+    customElement = document.createElement('pf-dropdown');
+    customElement.innerHTML = '<button class="btn btn-primary dropdown-toggle" type="button" id="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"> Dropdown<span class="caret"></span></button><ul class="dropdown-menu"><li class="dropdown-header">Header</li><li><a href="#">Item 1</a></li><li><a href="#">Item 2</a></li><li class="disabled"><a href="#">Item 3</a></li><li><a href="#">Item 4</a></li><li class="divider"></li><li><a href="#">Item 5</a></li></ul>';
+    document.body.appendChild(customElement);
+    if (customElement.initialized) {
+      done();
+    } else {
+      //wait for initialization
+      customElement.addEventListener('initialized', function (event) {
+        done();
+      });
+    }
+  });
+
+  afterEach(function () {
+    document.body.removeChild(customElement);
+  });
+
+  it("should open if closed or close if open the dropdown on toggle()", function (done) {
+    var button = customElement.querySelector('#button');
+    //open dropdown
+    customElement.toggle();
+    customElement.addEventListener('shown.bs.dropdown', function () {
+      expect(button.parentNode.classList.contains('open')).toBe(true);
+
+      //close dropdown
+      customElement.toggle();
+    }, false);
+
+    customElement.addEventListener('hidden.bs.dropdown', function () {
+      expect(button.parentNode.classList.contains('open')).toBe(false);
+    }, false);
+    done();
+  });
+
+  it("disabled button should not open dropdown", function () {
+    var button = customElement.querySelector('.btn');
+    button.classList.add('disabled');
+    customElement.toggle();
+    expect(button.parentNode.classList.contains('open')).toBe(false);
+    button.dispatchEvent(new MouseEvent('click'));
```

Comment:
I'd still like to see this can be rewritten without dispatching an event yourself. Similar to the tests you've written below. We should be able to call click() on the button element.